components/role-mgt/org.wso2.carbon.identity.role.v2.mgt.core/src/test/java/org/wso2/carbon/identity/role/v2/mgt/core/dao/RoleDAOTest.java

```diff
@@ -520,6 +570,91 @@ public void testUpdateRoleName() throws Exception {
+        mockCacheClearing(roleDAO);
+        identityDatabaseUtil.when(() -> IdentityDatabaseUtil.getUserDBConnection(anyBoolean()))
+                .thenAnswer(invocation -> getConnection());
+        identityDatabaseUtil.when(() -> IdentityDatabaseUtil.getDBConnection(anyBoolean()))
+                .thenAnswer(invocation -> getConnection());
+        identityUtil.when(IdentityUtil::getPrimaryDomainName).thenReturn(USER_DOMAIN_PRIMARY);
+        identityUtil.when(() -> IdentityUtil.extractDomainFromName(anyString())).thenCallRealMethod();
+        userCoreUtil.when(() -> UserCoreUtil.isEveryoneRole(anyString(), any(RealmConfiguration.class)))
+                .thenReturn(false);
+        identityTenantUtil.when(() -> IdentityTenantUtil.getTenantId(anyString())).thenReturn(SAMPLE_TENANT_ID);
+        userCoreUtil.when(() -> UserCoreUtil.removeDomainFromName(anyString())).thenCallRealMethod();
+        // Constructing a shared role scenario
+        RoleBasicInfo roleBasicInfo = addRole("sharing-org-role-200", ORGANIZATION_AUD, SAMPLE_ORG_ID, roleDAO);
+        roleDAO.getRoleAudienceRefId(roleBasicInfo.getId(), SAMPLE_TENANT_DOMAIN);
+        RoleBasicInfo sharedRoleBasicInfo = addRole("sharing-org-role-200", ORGANIZATION_AUD,
+                SAMPLE_TENANT_DOMAIN_TEST_ORG_ID, roleDAO);
+        roleDAO.addMainRoleToSharedRoleRelationship(roleBasicInfo.getId(), sharedRoleBasicInfo.getId(),
+                SAMPLE_TENANT_DOMAIN, SAMPLE_TENANT_DOMAIN_TEST);
+        addRole("sub-org-role-200", ORGANIZATION_AUD, SAMPLE_TENANT_DOMAIN_TEST_ORG_ID, roleDAO);
+        // Updating role name of a shared organization audience role
+        OrganizationManager organizationManager = mock(OrganizationManager.class);
+        RoleManagementServiceComponentHolder.getInstance().setOrganizationManager(organizationManager);
+        lenient().when(organizationManager.resolveOrganizationId(anyString())).thenReturn(
+                SAMPLE_TENANT_DOMAIN_TEST_ORG_ID);
+        identityTenantUtil.when(() -> IdentityTenantUtil.getTenantId(SAMPLE_TENANT_DOMAIN_TEST)).thenReturn(2);
+        identityTenantUtil.when(() -> IdentityTenantUtil.getTenantDomain(anyInt()))
+                .thenReturn(SAMPLE_TENANT_DOMAIN_TEST);
+
+        // Mocking isExistingRoleName method to return true for the first call and false for the second call
+        AtomicInteger callCount = new AtomicInteger(0);
+        doAnswer(invocation -> {
+            if (callCount.incrementAndGet() == 2) {
+                return true;
+            }
+            return false;
+        }).when(roleDAO).isExistingRoleName(anyString(), anyString(), anyString(), anyString());
+        MockedStatic<LoggerUtils> loggerUtils = mockStatic(LoggerUtils.class);
+        loggerUtils.when(LoggerUtils::isEnableV2AuditLogs).thenReturn(true);
+        roleDAO.updateRoleName(roleBasicInfo.getId(), "sub-org-role-200", SAMPLE_TENANT_DOMAIN);
+        loggerUtils.close();
```

Comment:
can you add a verifier to make sure that buildAuditData function is invoked

Reply:
Added an assertion to the WARN log. `buildAuditData` is a private method.